compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -951,45 +1040,31 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             Ok(())
         };
 
-        // Cleanup: deallocate locals.
-        // Usually we want to clean up (deallocate locals), but in a few rare cases we don't.
-        // We do this while the frame is still on the stack, so errors point to the callee.
-        let return_to_block = self.frame().return_to_block;
-        let cleanup = match return_to_block {
-            StackPopCleanup::Goto { .. } => true,
-            StackPopCleanup::Root { cleanup, .. } => cleanup,
-        };
-        if cleanup {
-            // We need to take the locals out, since we need to mutate while iterating.
-            let locals = mem::take(&mut self.frame_mut().locals);
-            for local in &locals {
-                self.deallocate_local(local.value)?;
-            }
-        }
-
         // All right, now it is time to actually pop the frame.
-        // Note that its locals are gone already, but that's fine.
-        let frame =
-            self.stack_mut().pop().expect("tried to pop a stack frame, but there were none");
+        let frame = self.pop_stack_frame(unwinding)?;
```

Comment:
This variable has type `StackPop`... please don't call it `frame`.

Reply:
Hm. Do you have ideas for the name? I see how `frame` can be misleading, but `stack_pop` is not very descriptive either...

Reply:
If in doubt I'd name it after the type it has. That's definitely better than `frame`.^^ If that name is not very descriptive then maybe the type needs to be renamed as well...

Reply:
Yeah, I don't love the type's name too...

Reply:
`stack_pop_action`/`stack_pop_info` would work for me. :shrug:  At least they are a lot better than `frame`.

Reply:
6eb672faeca6c20c9bed2ceff92996f791173ac7

Although `Info` does sound like nothing to me...